workitem/workitem_repository.go

```diff
@@ -1131,3 +1161,191 @@ func (r *GormWorkItemRepository) LoadByIteration(ctx context.Context, iterationI
+					}
+				}
+				tempList = append(tempList, val)
+			}
+			// Convert []string to comma seperated strings.
+			textToPrepend += fmt.Sprintf("\n\n %s : %s", oldWIType.Fields[fieldName].Label, strings.Join(tempList, ", "))
+		}
+		textToPrepend += "\n\n================================================\n\n"
+		wiStorage.Fields[SystemDescription] = *(rendering.NewMarkupContentFromValue(textToPrepend + originalDescription.Content))
+	}
+	// Set default values for all field in newWIType
+	for fieldName, fieldDef := range newWIType.Fields {
+		fieldValue := wiStorage.Fields[fieldName]
+		// Do not assign default value to metastate
+		if fieldName == SystemMetaState {
+			continue
+		}
+		// Assign default only if fieldValue is nil
+		wiStorage.Fields[fieldName], err = fieldDef.ConvertToModel(fieldName, fieldValue)
+		if err != nil {
+			return errs.Wrapf(err, "failed to convert field %s", fieldName)
+		}
+	}
+	wiStorage.Type = newWIType.ID
+	return nil
+}
+
+// getValueOfRelationKind resolved the relational value stored in val to it's verbose value. Eg: UUID of kind User to username.
+func getValueOfRelationalKind(db *gorm.DB, val interface{}, kind Kind) (string, error) {
+	var result string
+	switch kind {
+	case KindList, KindEnum:
+		return result, errors.NewInternalErrorFromString("cannot resolve relational value for KindList or KindEnum")
+	case KindUser:
+		var identity account.Identity
+		tx := db.Model(&account.Identity{}).Where("id = ?", val).Find(&identity)
+		if tx.Error != nil {
+			return result, errs.Wrap(tx.Error, "failed to find identity")
+		}
+		var user account.User
+		tx = db.Model(&account.User{}).Where("id = ?", identity.UserID).Find(&user)
+		if tx.Error != nil {
+			return result, errs.Wrap(tx.Error, "failed to find user")
+		}
+		result = fmt.Sprintf("%s (%s)", user.FullName, identity.Username)
+	case KindArea:
+		var area area.Area
+		tx := db.Model(area.TableName()).Where("id = ?", val).First(&area)
+		if tx.Error != nil {
+			return result, errs.Wrap(tx.Error, "failed to find area")
+		}
+		result = fmt.Sprintf("%s (%s)", area.Name, area.Path)
+	case KindBoardColumn:
+		var column BoardColumn
+		tx := db.Model(column.TableName()).Where("id = ?", val).First(&column)
+		if tx.Error != nil {
+			return result, errs.Wrap(tx.Error, "failed to find boardcolumn")
+		}
+		result = column.Name
+
+	case KindIteration:
+		var iteration iteration.Iteration
+		tx := db.Model(iteration.TableName()).Where("id = ?", val).First(&iteration)
+		if tx.Error != nil {
+			return result, errs.Wrap(tx.Error, "failed to find iteration")
+		}
+		result = fmt.Sprintf("%s (%s)", iteration.Name, iteration.Path)
+
+	case KindCodebase:
+		var codebase codebase.Codebase
+		tx := db.Model(codebase.TableName()).Where("id = ?", val).First(&codebase)
+		if tx.Error != nil {
+			return result, errs.Wrap(tx.Error, "failed to find codebase")
+		}
+		result = codebase.URL // TODO: Figure out what we should be here. Codebase does not have a name.
```

Comment:
@jarifibrahim Please change `// TODO:` to `// TODO(jarifibrahim):` then we all know who wrote the To Do and who to ask for it.

Reply:
Done in https://github.com/fabric8-services/fabric8-wit/pull/2202/commits/076caefddd986deeb6558786b34ad121028a28e0